### DISCUSSION TITLE
Uyuni with Google Oauth

### DISCUSSION THREAD

**Discussion by rchekhina** [uyuni-project/uyuni#7261] [Ideas]
HI,

is it possible to inegrate uyuni with Google Oauth to login with gmail account and manage permission with gmail groups ?

Thanks in advance.

Best regards.

**Comment by rchekhina** [comment]
HI,
To connect uyuni to google and can login do the following steps:

- Go to https://admin.google.com/
- Go to Apps, Web and mobile apps, click on Add app and select Add custom SAML app:
![image](https://github.com/uyuni-project/uyuni/assets/78473404/bd4e769a-6774-4f50-8f86-8cb051ec6680)
- Give it a name, upload the uyuni logo and click on CONTINUE:
![image](https://github.com/uyuni-project/uyuni/assets/78473404/2f2bf6f4-d1e3-4281-aa0d-d8f65699b96b)
- Copy the SSO URL, Entity ID and certificate we will use them on the /etc/rhn/rhn.conf file and click on Continue:
![image](https://github.com/uyuni-project/uyuni/assets/78473404/ed165ffa-7fbd-4d87-81e7-4eeeb6ea2091)
- On ACS URL set the url of uyuni with path /rhn/manager/sso/acs, on Entity ID set the url of uyuni with path /rhn/manager/sso/metadata, on Start URL set the uyuni url, select the box Signed response, on Name ID Format select UNSPECIFIED, on Name ID select Basic Information > Primary email and click on CONTINUE:
![image](https://github.com/uyuni-project/uyuni/assets/78473404/0e3cb557-a25f-4770-8d4d-fb8772aedcb7)
- Add the attribute mapping Primary email > uid and click on FINISH:
![image](https://github.com/uyuni-project/uyuni/assets/78473404/74e43e8b-a962-468b-bdaf-75b090263023)
- Click on User access:
![image](https://github.com/uyuni-project/uyuni/assets/78473404/af3a6408-d2a9-474c-8af6-53fdc76a39c9)
- Click on ON for everyone and click on SAVE:
![image](https://github.com/uyuni-project/uyuni/assets/78473404/b642781a-546b-49e0-94a5-1604155857c7)
- On Uyuni you need to create all the user locally with their gmail email to map them to google. the username need to be the gmail account, you dont need to set the password and you must select use PAM and fill out the Account information with the same gmail account like following:
![image](https://github.com/uyuni-project/uyuni/assets/78473404/1434c4f6-9354-4bde-85f8-1aa2d14f36d6)
- On the uyuni server edit the file /etc/rhn/rhn.conf and add the following:
```
java.sso = true

java.sso.onelogin.saml2.sp.entityid                       = https://DNS-UYUNI/rhn/manager/sso/metadata
java.sso.onelogin.saml2.sp.assertion_consumer_service.url = https://DNS-UYUNI/rhn/manager/sso/acs
java.sso.onelogin.saml2.sp.single_logout_service.url      = https://DNS-UYUNI/rhn/manager/sso/sls



java.sso.onelogin.saml2.idp.entityid  = Entity ID that you have copied 
java.sso.onelogin.saml2.idp.single_sign_on_service.url = SSO URL that you have copied 
java.sso.onelogin.saml2.idp.single_logout_service.url  = https://DNS-UYUNI/rhn/manager/sso/sls

# paste the certificate that you have copied in oneline
java.sso.onelogin.saml2.idp.x509cert = -----BEGIN CERTIFICATE----- 
```
- restart uyuni:
`spacewalk-service restart`
- IIf you want to disable the SSO you need to edit the file /etc/rhn/rhn.conf and change java.sso from true to false and restart uyuni

**Reply by juliogonzalez** [reply]
Very nice!

Can I move this to the Wiki @rchekhina?

**Reply by rchekhina** [reply]
HI Julio,
yes you can. Only one thing this parameter java.sso.onelogin.saml2.idp.single_logout_service.url  = https://DNS-UYUNI/rhn/manager/sso/sls is not the best because it doesnt close the gmail session I put it like this because I for the moment gsuite doesnt support the single logout service so I put the uyuni logout.
